src/prefect/storage/bitbucket.py

```diff
@@ -167,3 +252,55 @@ def _get_bitbucket_client(self) -> "atlassian.Bitbucket":
         else:
             session.headers["Authorization"] = "Bearer " + access_token
         return Bitbucket(host, session=session)
+
+    def _get_bitbucket_cloud_client(self) -> "atlassian.Bitbucket":
+        try:
+            from atlassian.bitbucket.cloud import Cloud
+        except ImportError as exc:
+            raise ImportError(
+                "Unable to import atlassian, please ensure you have installed the bitbucket extra"
+            ) from exc
+
+        if self.cloud_username_secret is not None:
+            # If cloud username secret specified, load it
+            cloud_username = Secret(self.cloud_username_secret).get()
+        else:
+            # Otherwise, fallback to loading from local secret or environment variable
+            cloud_username = prefect.context.get("secrets", {}).get(
+                "BITBUCKET_CLOUD_USERNAME"
+            )
+            if cloud_username is None:
+                cloud_username = os.getenv("BITBUCKET_CLOUD_USERNAME")
+
+        if self.cloud_app_password_secret is not None:
+            # If cloud app password secret specified, load it
+            cloud_app_password = Secret(self.cloud_app_password_secret).get()
+        else:
+            # Otherwise, fallback to loading from local secret or environment variable
+            cloud_app_password = prefect.context.get("secrets", {}).get(
+                "BITBUCKET_CLOUD_APP_PASSWORD"
+            )
```

Comment:
This is handled by the `Secret` class and depends on a config flag (`use_local_secrets`). I think you should remove this fallback and the one for the username. I believe the `os.getenv` fallback is still in-line with our typical pattern for something like this so that's okay.

Reply:
I've removed this and the one for the username. 

I had copied this from _get_bitbucket_server_client() (which I didn't write and have not changed)

On our infrastructure (Prefect Server) we will be relying on `os.getenv`

Reply:
Ah sorry about that -- looks like it might just be old pattern there.

Reply:
Shall I remove it from the other function?

Reply:
I can do it in a follow-up PR. I think it's better to isolate a change to the existing bit.

Reply:
That makes sense.